apps/wallet-dashboard/lib/utils/migration/groupStardustObjectsByMigrationStatus.ts

```diff
@@ -149,3 +162,18 @@ export function extractNotOwnedStorageDepositReturnAmount(
     }
     return null;
 }
+
+export function calculateTotalNativeTokensByName(objects: ResolvedObjectTypes[]): number {
+    const uniqueTokens: string[] = [];
+
+    return objects.reduce((total, obj) => {
+        if (
+            obj.commonObjectType === CommonMigrationObjectType.NativeToken &&
+            !uniqueTokens.includes(obj.name)
```

Comment:
I think there are be potentially 2 coins with the same name but the type must be different, so we should check against the type, not the name

Reply:
eg 
```ts
export function getUniqueNativeTokensByCoinType(objects: ResolvedObjectTypes[]): number {
    return new Set(
        objects
            .filter((obj) => obj.commonObjectType === CommonMigrationObjectType.NativeToken)
            .map((obj) => obj.coinType),
    ).size;
}
```